stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/TextAndAttachmentsViewHolder.kt

```diff
@@ -80,10 +86,25 @@ internal class TextAndAttachmentsViewHolder(
     }
 
     private fun setupAttachment(data: MessageListItem.MessageItem) {
-        with(binding.attachmentsContainer) {
-            removeAllViews()
-            addView(attachmentViewFactory.createAttachmentView(data, listeners, style, binding.root))
+        attachmentViewHolderFactory.setUp(data.message.attachments)
+        adapter = FileAttachmentsAdapter(attachmentViewHolderFactory)
+
+        binding.run {
+            attachmentsRecycler.layoutManager = LinearLayoutManager(context)
```

Comment:
This might be useful to set somewhere around here https://developer.android.com/reference/androidx/recyclerview/widget/LinearLayoutManager#setRecycleChildrenOnDetach(boolean)